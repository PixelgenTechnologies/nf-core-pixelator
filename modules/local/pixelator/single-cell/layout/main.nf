process PIXELATOR_LAYOUT {
    tag "$meta.id"
    label 'process_medium'


    conda "bioconda::pixelator=0.17.1"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/pixelator:0.17.1--pyhdfd78af_0' :
        'biocontainers/pixelator:0.17.1--pyhdfd78af_0' }"

    input:
    tuple val(meta), path(data)

    output:
    tuple val(meta), path("layout/*dataset.pxl")  , emit: dataset
    tuple val(meta), path("layout/*report.json")  , emit: report_json
    tuple val(meta), path("layout/*.meta.json")   , emit: metadata
    tuple val(meta), path("layout/*")             , emit: all_results
    tuple val(meta), path("*pixelator-layout.log"), emit: log

    path "versions.yml"                           , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:

    prefix = task.ext.prefix ?: "${meta.id}"
    def args = task.ext.args ?: ''

    """
    pixelator \\
        --cores $task.cpus \\
        --log-file ${prefix}.pixelator-layout.log \\
        --verbose \\
        single-cell \\
        layout \\
        --output . \\
        $args \\
        $data

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        pixelator: \$(echo \$(pixelator --version 2>/dev/null) | sed 's/pixelator, version //g' )
    END_VERSIONS
    """
}
