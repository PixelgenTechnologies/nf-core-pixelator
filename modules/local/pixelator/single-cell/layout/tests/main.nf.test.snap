{
    "pxl": {
        "content": [
            "metadata.json:md5,f2fb857ccf3ee1b600c6903b7406444d"
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-01-14T13:53:40.923684802"
    },
    "Test MPX layout - SCSP v1 | Immunology-I": {
        "content": [
            [
                [
                    {
                        "id": "sample01_1k_pbmcs_scsp_v1_immunology1",
                        "design": "D21",
                        "panel": "human-sc-immunology-spatial-proteomics-1",
                        "technology": "mpx"
                    },
                    "sample01_1k_pbmcs_scsp_v1_immunology1.report.json:md5,479afd25dcacc45540d81d1d8b7b5346"
                ]
            ],
            [
                [
                    {
                        "id": "sample01_1k_pbmcs_scsp_v1_immunology1",
                        "design": "D21",
                        "panel": "human-sc-immunology-spatial-proteomics-1",
                        "technology": "mpx"
                    },
                    "sample01_1k_pbmcs_scsp_v1_immunology1.meta.json:md5,64e167effed9d3160c1d9f2964538c2a"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-01-14T13:53:40.903975567"
    },
    "Test MPX layout - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "sample01_1k_pbmcs_scsp_v1_immunology1",
                            "design": "D21",
                            "panel": "human-sc-immunology-spatial-proteomics-1",
                            "technology": "mpx"
                        },
                        "sample01_1k_pbmcs_scsp_v1_immunology1.layout.dataset.pxl:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "sample01_1k_pbmcs_scsp_v1_immunology1",
                            "design": "D21",
                            "panel": "human-sc-immunology-spatial-proteomics-1",
                            "technology": "mpx"
                        },
                        "sample01_1k_pbmcs_scsp_v1_immunology1.report.json:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "sample01_1k_pbmcs_scsp_v1_immunology1",
                            "design": "D21",
                            "panel": "human-sc-immunology-spatial-proteomics-1",
                            "technology": "mpx"
                        },
                        "sample01_1k_pbmcs_scsp_v1_immunology1.meta.json:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "3": [
                    [
                        {
                            "id": "sample01_1k_pbmcs_scsp_v1_immunology1",
                            "design": "D21",
                            "panel": "human-sc-immunology-spatial-proteomics-1",
                            "technology": "mpx"
                        },
                        [
                            "sample01_1k_pbmcs_scsp_v1_immunology1.layout.dataset.pxl:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "sample01_1k_pbmcs_scsp_v1_immunology1.meta.json:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "sample01_1k_pbmcs_scsp_v1_immunology1.report.json:md5,d41d8cd98f00b204e9800998ecf8427e"
                        ]
                    ]
                ],
                "4": [
                    [
                        {
                            "id": "sample01_1k_pbmcs_scsp_v1_immunology1",
                            "design": "D21",
                            "panel": "human-sc-immunology-spatial-proteomics-1",
                            "technology": "mpx"
                        },
                        "sample01_1k_pbmcs_scsp_v1_immunology1.pixelator-layout.log:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "5": [
                    "versions.yml:md5,a6eae3a2fec653d9668e392abeafe825"
                ],
                "all_results": [
                    [
                        {
                            "id": "sample01_1k_pbmcs_scsp_v1_immunology1",
                            "design": "D21",
                            "panel": "human-sc-immunology-spatial-proteomics-1",
                            "technology": "mpx"
                        },
                        [
                            "sample01_1k_pbmcs_scsp_v1_immunology1.layout.dataset.pxl:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "sample01_1k_pbmcs_scsp_v1_immunology1.meta.json:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "sample01_1k_pbmcs_scsp_v1_immunology1.report.json:md5,d41d8cd98f00b204e9800998ecf8427e"
                        ]
                    ]
                ],
                "dataset": [
                    [
                        {
                            "id": "sample01_1k_pbmcs_scsp_v1_immunology1",
                            "design": "D21",
                            "panel": "human-sc-immunology-spatial-proteomics-1",
                            "technology": "mpx"
                        },
                        "sample01_1k_pbmcs_scsp_v1_immunology1.layout.dataset.pxl:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "log": [
                    [
                        {
                            "id": "sample01_1k_pbmcs_scsp_v1_immunology1",
                            "design": "D21",
                            "panel": "human-sc-immunology-spatial-proteomics-1",
                            "technology": "mpx"
                        },
                        "sample01_1k_pbmcs_scsp_v1_immunology1.pixelator-layout.log:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "metadata": [
                    [
                        {
                            "id": "sample01_1k_pbmcs_scsp_v1_immunology1",
                            "design": "D21",
                            "panel": "human-sc-immunology-spatial-proteomics-1",
                            "technology": "mpx"
                        },
                        "sample01_1k_pbmcs_scsp_v1_immunology1.meta.json:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "report_json": [
                    [
                        {
                            "id": "sample01_1k_pbmcs_scsp_v1_immunology1",
                            "design": "D21",
                            "panel": "human-sc-immunology-spatial-proteomics-1",
                            "technology": "mpx"
                        },
                        "sample01_1k_pbmcs_scsp_v1_immunology1.report.json:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,a6eae3a2fec653d9668e392abeafe825"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-01-14T13:53:18.727581453"
    }
}